Source/com/drew/metadata/exif/ExifReader.java

```diff
@@ -72,29 +75,22 @@ public static boolean startsWithJpegExifPreamble(byte[] bytes)
             new String(bytes, 0, JPEG_SEGMENT_PREAMBLE.length()).equals(JPEG_SEGMENT_PREAMBLE);
     }
 
-    /** Reads TIFF formatted Exif data from start of the specified {@link RandomAccessReader}. */
-    public void extract(@NotNull final RandomAccessReader reader, @NotNull final Metadata metadata)
-    {
-        extract(reader, metadata, 0);
-    }
-
     /** Reads TIFF formatted Exif data a specified offset within a {@link RandomAccessReader}. */
-    public void extract(@NotNull final RandomAccessReader reader, @NotNull final Metadata metadata, int readerOffset)
+    public void extract(@NotNull final RandomAccessReader reader, @NotNull final Metadata metadata)
     {
-        extract(reader, metadata, readerOffset, null);
+        extract(reader, metadata, null);
     }
 
     /** Reads TIFF formatted Exif data at a specified offset within a {@link RandomAccessReader}. */
-    public void extract(@NotNull final RandomAccessReader reader, @NotNull final Metadata metadata, int readerOffset, @Nullable Directory parentDirectory)
+    public void extract(@NotNull final RandomAccessReader reader, @NotNull final Metadata metadata, @Nullable Directory parentDirectory)
     {
-        ExifTiffHandler exifTiffHandler = new ExifTiffHandler(metadata, parentDirectory, readerOffset);
+        ExifTiffHandler exifTiffHandler = new ExifTiffHandler(metadata, parentDirectory, /*readerOffset*/ 0); // FIXME what to do?
```

Comment:
@drewnoakes I've rebase this PR and faced conflict with #578 . I have no idea what to do here, can you please help me?

Reply:
@don-vip I pushed a commit which does what I think is needed, based on the .NET implementation. They're quite similar implementations. I haven't done much testing with it, so please do verify. Thanks for revisiting this!

Reply:
I've checked, I don't see anything suspicious in the images diff, but it's quite difficult to be sure as there are a lot of changes that do not seem related at all to this PR, plus the date changes caused by the use of Sydney timezone. I managed at least to make the diff not language-dependent, see https://github.com/drewnoakes/metadata-extractor-images/pull/52